London | 25-SDC-July | Fatma Arslantas | Sprint 3 | Implement Shell Tools

AFatmaa opened the pull request:
<!--

You must title your PR like this:

Region | Cohort | FirstName LastName | Sprint | Assignment Title

For example,

London | 25-ITP-May | Carol Owen | Sprint 1 | Alarm Clock

Complete the task list below this message.
If your PR is rejected, check the task list.

-->

## Learners, PR Template

Self checklist

- [ ] I have committed my files one by one, on purpose, and for a reason
- [ ] I have titled my PR with Region | Cohort | FirstName LastName | Sprint | Assignment Title
- [ ] I have tested my changes
- [ ] My changes follow the [style guide](https://curriculum.codeyourfuture.io/guides/reviewing/style-guide/)
- [ ] My changes meet the [requirements](./README.md) of this task

## Changelist

This pull request implements the `cat`, `wc`, and `ls` shell tools in JavaScript using NodeJS, as per the coursework instructions.

I tried to make them work like the real tools for the tasks given in the README.md files.

## Questions

Ask any questions you have for your reviewer.

AFatmaa committed:
Set up package.json and install Commander

- Added "type": "module" to package.json
- Installed commander as a dependency

AFatmaa committed:
Implement cat command

AFatmaa committed:
Implement ls command

AFatmaa committed:
Implement wc command

LonMcGregor reviewed (commented):
Hi,

First, I notice you haven't updated the template in your pull request. when opening a new PR on github, remember to add in your own explanation in the text.

You have made a good start implementing the tasks. I have left some comments to point you towards how you could improve the more.

AFatmaa committed:
fix: Align line numbers

AFatmaa commented on implement-shell-tools/cat/cat.js:
```diff
@@ -0,0 +1,36 @@
+
+  for (const line of content.split("\n")) {
+    if (options.n) {
+      console.log(`${lineNumber}  ${line}`);
```
Thank you for the feedback and your time. You're right, numbers like '10' took up more space and pushed the text. I've fixed this by using `padStart`, and I updated my code.

AFatmaa committed:
feat: Show files in one line if -1 option is not used

AFatmaa commented on implement-shell-tools/ls/ls.js:
```diff
@@ -0,0 +1,24 @@
+program
+  .name("my-ls")
+  .description("Reimplementation of the Unix `ls` command with -1 and -a options")
+  .option("-1", "list one file per line")
```
Thanks for the feedback! You're right. I updated the code so that if -1 is not passed, the files will now be shown in a single line, just like the default ls command.

AFatmaa committed:
refactor: Simplify output formatting and fix line count calculation

AFatmaa commented:
Hi @LonMcGregor,
Thank you so much for your guidance and your time! I’ve updated the PR template with an explanation. I hadn’t realised I needed to fill it in before, but I’ll make sure to do it properly for future PRs. I’ve also updated my code based on your comments. Could you please check them when you have time? Thanks again! 🙏
